exposestrategy/ingress.go

```diff
@@ -278,18 +246,69 @@ func (s *IngressStrategy) Add(svc *api.Service) error {
 		}
 	}
 
-	if createIngress {
-		_, err := s.client.Ingress(ingress.Namespace).Create(ingress)
+	existing, err := cli.Ingress(svc.Namespace).Get(appName)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return "", errors.Wrapf(err, "could not check for existing ingress %s/%s", svc.Namespace, appName)
+		}
+		_, err := cli.Ingress(ingress.Namespace).Create(ingress)
 		if err != nil {
-			return errors.Wrapf(err, "failed to create ingress %s/%s", ingress.Namespace, ingress.Name)
+			return "", errors.Wrapf(err, "failed to create ingress %s/%s", ingress.Namespace, ingress.Name)
 		}
+	} else if existing.Annotations["fabric8.io/generated-by"] != "exposecontroller" {
+		return "", errors.Errorf("ingress %s/%s already exists", svc.Namespace, appName)
 	} else {
-		_, err := s.client.Ingress(svc.Namespace).Update(ingress)
+		update := false
+		if existing.Labels == nil {
+			existing.Labels = map[string]string{}
+		}
+		for k, v := range ingress.Labels {
+			if w, ok := existing.Labels[k]; !ok || v != w {
+				update = true
+				existing.Labels[k] = v
+			}
+		}
+		if !reflect.DeepEqual(ingress.Annotations, existing.Annotations) {
```

Comment:
have you considered alternatives to deepequal?

Reply:
I do not know of any alternatives 😄  Care to enlighten me?